base.py

```diff
@@ -1,20 +1,28 @@
 from abc import ABC, abstractmethod
 
-import joblib
 from joblib import delayed, effective_n_jobs, Parallel
 from typing import List, Union
 
 import numpy as np
 import pandas as pd
+from scipy.sparse import vstack, csr_array
 from sklearn.base import BaseEstimator, TransformerMixin
 
 from rdkit.Chem import MolFromSmiles
 from rdkit.Chem.rdchem import Mol
 
+legal_result_types = {"bit", "sparse", "count", "sparse_count"}
```

Comment:
Use UPPER_CASE for global variables

Reply:
Also, maybe use `Enum`?

Reply:
Also, this probably doesn't need to be a global variable

Reply:
variable removed. the check is made once anyway so it will be performed explicitly as 
`assert result_vector_type in ["bit", "sparse", "count", "sparse_count"]`